### PR TITLE
Fix #171

### LINKED ISSUE #171: basil/oxford_asl --bolus parameter can accept a comma-separated list
Hi, I have a use-case related to the `oxford_asl` command (which the [`BASILCBF`]() interface wraps). I have some multi-delay ASL data which has the M0 image(s) included.

Since the data is multi-delay (and because the M0 is included) the `PostLabelingDelay` metadata field is an array of the form:

```text
"PostLabelingDelay": [0.0, 0.0, pld1, pld2, pld3, ...]
                      ^^^^^^^^
                      M0 images
```

As described in the BIDS spec for [`PostLabelingDelay`](https://bids-specification.readthedocs.io/en/v1.6.0/04-modality-specific-files/01-magnetic-resonance-imaging-data.html#common-metadata-fields-applicable-to-both-pcasl-and-pasl) the M0 images appear as 0's (above).

Similarly, the `LabelingDuration` is also an array where we play the same game with the M0 Images, i.e. an array of the form:
```text
"LabelingDuration": [0.0, 0.0, ld1, ld2, ld3, ...]
                     ^^^^^^^^
                     M0 images
```

It turns out that the `--bolus` parameter for the `oxford_asl` command can be either a number or a list of numbers.

**tl;dr: I was wondering if the `bolus` parameter of the [`BASILCBF`]() interface could/should be made to accept either a `float` or `list[float]`.**

### PR DESCRIPTION
**Pull request by andrewrosss**
This is a potential fix for #171.

Basically, the [`tis`](https://github.com/PennLINC/aslprep/blob/b9f9ddc28d9af1e97ca31da72aa76fabaf4b68dc/aslprep/interfaces/cbf_computation.py#L724) and [`bolus`](https://github.com/PennLINC/aslprep/blob/b9f9ddc28d9af1e97ca31da72aa76fabaf4b68dc/aslprep/interfaces/cbf_computation.py#L727) inputs defined in the [`_BASILCBFInputSpec`](https://github.com/PennLINC/aslprep/blob/b9f9ddc28d9af1e97ca31da72aa76fabaf4b68dc/aslprep/interfaces/cbf_computation.py#L709) have been converted to traits of the form:

```python
taits.Either(
    traits.Float(),
    traits.List(traits.Float()),
    ...
)
```

This allows those parameters to be specified as either a scalar value (float) or a list of floats, and the trait itself will handle to conversion to a comma-separated list of numbers (string). To get a sense of what this particular change would do here is a sample script showing what happens to the resulting `oxford_asl` commands:

```python
# sample.py
import tempfile
import warnings
from pathlib import Path

warnings.filterwarnings("ignore")

import aslprep.interfaces.cbf_computation as cbfcomp


def main() -> int:
    with tempfile.TemporaryDirectory() as _tmpdir:
        tmpdir = Path(_tmpdir)

        f = create_a_real_file(tmpdir)  # so the traits.File attributes don't complain

        # BASILCBF with float --bolus and --tis
        basil1 = cbfcomp.BASILCBF(
            in_file=f,
            mask=f,
            m0scale=0.0,
            m0tr=0.0,
            tis=1.11,  # float
            bolus=4.44,  # float
            out_basename='out_base'
        )

        print('SCALARS', basil1.cmdline, sep='\n')
        assert ' --tis 1.11 ' in basil1.cmdline
        assert ' --bolus 4.44 ' in basil1.cmdline

        # BASILCBF with list[float] --bolus and --tis
        basil2 = cbfcomp.BASILCBF(
            in_file=f,
            mask=f,
            m0scale=0.0,
            m0tr=0.0,
            tis=[1.11, 2.22, 3.33],  # plain old python list
            bolus=[4.4, 5.5, 6.6],  # plain old python list
            out_basename='out_base'
        )

        print('LISTS', basil2.cmdline, sep='\n')
        assert ' --tis 1.11,2.22,3.33 ' in basil2.cmdline
        assert ' --bolus 4.4,5.5,6.6 ' in basil2.cmdline

    return 0



def create_a_real_file(d: Path) -> Path:
    existing_file = d / 't.txt'
    existing_file.write_text('')
    return existing_file


if __name__ == "__main__":
    raise SystemExit(main())
```

Assuming this branch is active and the script is saved as `sample.py`, it can be run with `python sample.py`

In a nutshell, this scripts prints the resulting `oxford_asl` command for the cases where tis/bolus is a float and a list of floats.

### CODE REVIEW

**Comment by a3sha2** [comment]
@andrewrosss  m0scale can only be >= 1 not zero, that will make CBF value to be NAN
